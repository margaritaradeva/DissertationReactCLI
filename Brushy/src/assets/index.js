// Re-export visuals/audio to import more conveniently from this folder
export const profile = require('./profilepic.png');
export const dogWholeBody = require('./dog_whole_body.png')
export const zoomedInDog = require('./zoomedInDog.png');

