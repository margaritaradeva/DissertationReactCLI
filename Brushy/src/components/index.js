import CustomButton from "./CustomButton";
import Input from "./Input";

export {
    CustomButton,
    Input
};