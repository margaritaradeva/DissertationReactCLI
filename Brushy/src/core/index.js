
import useGlobally from "./global"
import { set, get,remove, wipe } from "./secure"
import utils from "./utils";
import api from "./api";
export {
    set,
    get,
    remove,
    wipe,
    utils,
    useGlobally,
    api
};