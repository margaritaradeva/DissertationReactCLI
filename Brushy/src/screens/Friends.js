// Import all of the necessary libraries, screens and components
// Will be implemented soon
import { StyleSheet, Text, View } from 'react-native';

export default function Friends() {
    return (
      <View style={styles.container}>
        <Text>Coming soon!</Text>
      </View>
    );
  }

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
    },
  });