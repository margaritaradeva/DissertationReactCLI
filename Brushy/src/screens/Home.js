import React, { useEffect, useRef, useState } from 'react';
import { StyleSheet, View, Animated, Button, Text, Modal, TouchableOpacity, TextInput, Platform, Dimensions } from 'react-native';
import { CountdownTimer } from '../components/common';
import { api, useGlobally } from '../core';
import { bathroom, bathroomZoomedIn } from '../assets/backgrounds';
import { BlurView } from '@react-native-community/blur';
import secure from '../core/secure';

export default function Home() {
  const { dogFullBody, userDetails, getDetails, setDogFullBodyImage } = useGlobally();
  const [showModalStart, setShowModalStart] = useState(false);
  const [value, setValue] = useState('');
  const zoomedInDog = require('../assets/zoomedInDog.png');

  // Object containing images for different levels
  const images = {
    dogFullBody: [
      require('../assets/mini_shop/level1_item.png'),
      require('../assets/mini_shop/level2_item.png'),
      // Add more level images here...
    ],
  };

  // Fetch user details when component mounts
  useEffect(() => {
    const fetchDetails = async () => {
      await getDetails();
    };

    fetchDetails();
  }, []);

  // Update dog image based on user details
  useEffect(() => {
    if (userDetails?.image_id) {
      const photoIndex = userDetails.image_id - 1;
      if (photoIndex >= 0 && photoIndex < images.dogFullBody.length) {
        const dogBodyImage = images.dogFullBody[photoIndex];
        setDogFullBodyImage(dogBodyImage);
        if (userDetails.is_char_name_set === false) {
          setShowModalStart(true);
        }
      }
    }
  }, [userDetails]);

  // Animation setup
  const fadeAnim = useRef(new Animated.Value(0)).current;

  const startAnimationAndTimer = () => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 3000,
      useNativeDriver: true,
    }).start();
  };

  const resetAnimation = () => {
    fadeAnim.setValue(0);
  };

  const bathroomOpacity = fadeAnim.interpolate({
    inputRange: [0, 1],
    outputRange: [1, 0],
  });

  const bathroomZoomedInOpacity = fadeAnim;

  // Function to set up character name
  async function SetUpName() {
    try {
      const credentials = await secure.get('credentials');
      const response = await api({
        method: 'POST',
        url: '/application/updateCharacterName/',
        data: {
          email: credentials.email,
          new_name: value,
        },
      });
      setShowModalStart(false);
    } catch (error) {
      console.log('error');
    }
  }

  return (
    <View style={styles.container}>
      {/* Display countdown timer */}
      <CountdownTimer seconds={120} startTimer={startAnimationAndTimer} onModalClose={resetAnimation} name={userDetails.character_name ? userDetails.character_name : 'there'} />

      {/* Background images and animations */}
      <View style={styles.backgroundContainer}>
        <Animated.Image source={bathroom} style={[styles.backgroundImage, { opacity: bathroomOpacity }]} />
        <Animated.Image source={bathroomZoomedIn} style={[styles.backgroundImage, { opacity: bathroomZoomedInOpacity }]} />
        <View style={styles.animationContainer}>
          {/* Main dog image */}
          <Animated.Image source={dogFullBody} style={[styles.image, { opacity: bathroomOpacity }]} resizeMode="contain" />
          {/* Zoomed-in dog image */}
          <Animated.Image source={zoomedInDog} style={[styles.image2, styles.absolutePosition, { opacity: bathroomZoomedInOpacity }]} resizeMode="contain" />
        </View>
      </View>

      {/* Modal for setting up character name */}
      <Modal visible={showModalStart} transparent={true} animationType="slide" onRequestClose={() => setShowModalStart(false)}>
        <BlurView style={StyleSheet.absoluteFill} blurType="light" blurAmount={10} reducedTransparencyFallbackColor="white" />
        <View style={styles.container}>
          <View style={styles.modalView}>
            <Text style={styles.infoText}>Pick a name for your character </Text>
            <View style={styles.modalButton}>
              <TextInput
                placeholder={'Character Name'}
                placeholderTextColor="gray"
                autoCapitalize="none"
                autoComplete="off"
                style={styles.inputText}
                value={value}
                onChangeText={(text) => {
                  setValue(text);
                }}
              />
              <TouchableOpacity style={styles.button} onPress={() => SetUpName()}>
                <Text style={styles.buttonText}>Set my name!</Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  backgroundContainer: {
    width: '100%',
    height: '100%',
    position: 'relative',
  },
  backgroundImage: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: 0,
    left: 0,
  },
  animationContainer: {
    // marginBottom: 0,
    // height: 300,
    // width: 300,
    position: 'absolute',
  },
  image: {
    width: 350,
    height: 350,
    marginTop: '58%',
    marginLeft:'12%'
  },
  absolutePosition: {
    position: 'absolute',
    top: 0,
    left: 0,
  },
  modalView: {
    margin:10,
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 35,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width:0,
      height:2,
    },
    shadowOpacity:0.25,
    shadowRadius: 3.84,
    elevation: 5,
    zIndex:1,
  },
  modalButton: {
    flexDirection: 'column',
    marginTop: 15,
    justifyContent: 'center',
    alignItems: 'center'
  },
  button: {
    backgroundColor: '#2E95CA',
    padding: 6,
    marginTop: 20,
    borderRadius: 10,
    borderColor: 'black',

    borderWidth: 2,
    justifyContent: 'center',
    alignItems: 'center',
    zIndex:1
  },
  buttonText: {
   
      color: '#FFFFFF',
      textShadowColor: '#000000',
      fontSize: 16,
      fontWeight: 'bold'
  
  },
  infoText: {
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    fontSize: 16,
    color: 'black'
  },
  image2: {
    width: 450,
    height: 450,
    marginTop: '44%',
    marginRight: '15%',
  },
  inputText: {
    backgroundColor:'white',
        height: Platform.OS == 'android' ? 40 : 20,
        paddingVertical: 0,
        borderColor: '#acacac',
        borderWidth:1,
        fontSize: 14,
        backgroundColor:'#f2f2f2',
        lineHeight: 14 ,
        color: 'black',
        borderRadius: 16, height:52, width:'100%',
         paddingHorizontal:16,fontSize:16
  }
});
