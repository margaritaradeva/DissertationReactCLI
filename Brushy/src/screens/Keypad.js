// Import necessary libraries and components
import { useEffect, useState } from "react";
import { StyleSheet, Text, TouchableOpacity, View, FlatList, Dimensions } from "react-native";
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'; // Icons
import { MotiView } from 'moti';
import { api } from "../core";
import { BlurView } from "@react-native-community/blur";
import { CustomButton, Input } from "../components/common";
import { Modal, TextInput } from "react-native";
import secure from "../core/secure";

// Define the keys for the pad
const digits = [1, 2, 3, 4, 5, 6, 7, 8, 9, '', 0, 'del'];

// Take the width of the screen
const { width } = Dimensions.get('window');

// Define the size of the digit circles
const circleSize = width * 0.2;

// Define the size of the text in the circles
const circleTextSize = circleSize * 0.4;

// Define pin length
const pinLength = 6;

// Define the size for block of text where the entered pin is displayed
const pinSize = width / 2;

// Define the size for each little square containing a digit
const pinTextSizeMax = pinSize / pinLength;
const pinTextSize = pinTextSizeMax - 10;

// PinPad component for displaying and interacting with the pin pad
function PinPad({ onPress }) {
    return (
        <FlatList
            numColumns={3}
            data={digits}
            style={{ flexGrow: 0 }}
            scrollEnabled={false}
            columnWrapperStyle={{ gap: 20 }}
            contentContainerStyle={{ gap: 20 }}
            keyExtractor={(_, index) => index.toString()}
            renderItem={({ item }) => {
                return (
                    <TouchableOpacity
                        disabled={item === ''}
                        onPress={() => onPress(item)}>
                        <View style={[styles.circles, { borderWidth: (item === '' || item === 'del') ? 0 : 1 }]}>
                            {item === 'del' ?
                                <MaterialCommunityIcons name="backspace-outline" color={'#000000'} size={circleTextSize} /> :
                                <Text style={styles.circleText}>{item}</Text>}
                        </View>
                    </TouchableOpacity>
                );
            }}
        />
    );
}

// Pin component for handling pin entry and pin setup
export default function Pin({ navigation }) {
    // State variables for pin entry and setup
    const [showModal, setShowModal] = useState(false);
    const [pin, setPin] = useState([]);
    const [firstPin, setFirstPin] = useState(null);
    const [confirmingPin, setConfirmingPin] = useState(false);
    const [loading, setLoading] = useState(true); // used for knowing if the async function has finished execution
    const [message, setMessage] = useState('');
    const [isPinSet, setIsPinSet] = useState(false);
    const [matchedPins, setMatchedPins] = useState(false);
    const [move, setMove] = useState(false);
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [emailError, setEmailError] = useState('');
    const [passwordError, setPasswordError] = useState('');
    const [requestError, setRequestError] = useState('');
    const [seePassword, setSeePassword] = useState(false);

    // Function to trigger pin animation
    const triggerAnimation = () => {
        setMatchedPins(true);
        setTimeout(() => {
            setMatchedPins(false);
            setMessage('Please set up your pin');
        }, 600);
    };

    // useEffect to check pin status when component renders
    useEffect(() => {
        checkPinStatus();
        setPin([]);
        setConfirmingPin(false);
    }, []);

    // useEffect to handle pin entry and setup
    useEffect(() => {
        // Check if pin setup is in progress
        if (!isPinSet) {
            // Handle pin setup
            if (pin.length === pinLength) {
                if (!confirmingPin) {
                    setFirstPin(pin);
                    setPin([]);
                    setConfirmingPin(true);
                } else {
                    const arePinsIdentical = firstPin.length === pin.length && firstPin.every((element, index) => element === pin[index]);
                    if (!arePinsIdentical) {
                        // Prompt users to enter both pins again if they weren't identical
                        setMessage('Pins do not match');
                        triggerAnimation();
                        setTimeout(() => {
                            setPin([]);
                            setFirstPin(null);
                            setConfirmingPin(false);
                        }, 600);
                    } else {
                        const pinString = firstPin.join('');
                        console.log(pinString);
                        setUpPin(pinString);
                    }
                }
            }
        } else {
            // Handle pin entry
            if (pin.length === pinLength) {
                const pinString = pin.join('');
                console.log(pinString);
                checkPin(pinString);
            }
        }
    }, [message, isPinSet, pin]);

    // useEffect to navigate after successful pin entry
    useEffect(() => {
        if (move) {
            setMessage('Please enter your pin');
            setPin([]);
            setMove(false);
            setMessage('Please enter your pin');
            navigation.navigate("Parents Settings");
        }
    }, [move]);

    // useEffect to update message based on pin status
    useEffect(() => {
        if (confirmingPin) {
            setMessage('Please confirm your pin');
        } else if (isPinSet) {
            setMessage('Please enter your pin');
        } else {
            setMessage('Please set up your pin');
        }
    }, [confirmingPin, isPinSet]);

    // Function to check pin status
    async function checkPinStatus() {
        try {
            const response = await api.get('/application/isPinSet/');
            setIsPinSet(response.data.is_pin_set);
            setLoading(false);
        } catch (error) {
            console.error("errorrrrr", error);
        }
        setLoading(false);
    }

    // Function to set up pin
    async function setUpPin(pinToSet) {
        try {
            const response = await api({
                method: 'POST',
                url: '/application/setPin/',
                data: {
                    "parent_pin": pinToSet
                }
            });
            setPin([]);
            setMessage('Please enter your pin');
            setMove(true);
            setIsPinSet(true);
            setConfirmingPin(false);
        } catch (error) {
            console.log(error);
        }
    }

    // Function to handle forgot pin
    async function forgotPin() {
        setIsPinSet(false);
    }

    // Function to check pin
    async function checkPin(pinToCheck) {
        try {
            const response = await api({
                method: 'POST',
                url: '/application/checkPin/',
                data: {
                    "parent_pin": pinToCheck
                }
            });
            if (response.status === 200) {
                console.log('yay');
                setPin([]);
                setMessage('Please enter your pin');
                setMove(true);
            }
        } catch (error) {
            console.log(error);
            setMessage('Pin is incorrect');
            triggerAnimation();
            setTimeout(() => {
                setPin([]);
            }, 600);
        }
    }

    // Function to handle pin pad button press
    const handlePress = (item) => {
        if (!isPinSet) {
            if (item !== '' && item !== 'del') {
                if (!confirmingPin && pin.length < pinLength) {
                    setPin(prevPin => [...prevPin, item]);
                } else if (confirmingPin && pin.length < pinLength) {
                    setPin(prevPin => [...prevPin, item]);
                }

            } else if (item === 'del') {
                setPin(prevPin => {
                    const newPin = prevPin.slice(0, prevPin.length - 1);
                    console.log(newPin);
                    return newPin;
                });
            }
        } else {
            if (item !== '' && item !== 'del') {
                if (pin.length < pinLength) {
                    setPin(prevPin => [...prevPin, item]);
                }
            } else if (item === 'del') {
                setPin(prevPin => {
                    const newPin = prevPin.slice(0, prevPin.length - 1);
                    return newPin;
                });
            }
        }
    };

    // Function to close modal
    async function closeModal() {
        setShowModal(false);
    }

    // Function to show modal on screen
    function showModalOnScreen() {
        setShowModal(true);
    }

    // Function to reset pin
    async function resetPIN() {
        // Clear Previous error
        setRequestError('');
        // Check email
        const noEmail = !email;
        if (noEmail) {
            setEmailError('Email is required');
        } else {
            const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
            if (!emailRegex.test(email)) {
                setEmailError('Please enter a valid email address');
            }
        }
        // Check password
        const noPassword = !password;
        if (noPassword) {
            setPasswordError('Password is required');
        }
        // Break out of this function if there were any issues
        if (noEmail || noPassword || validationErrors) {
            return;
        }
        try {
            const response = await api({
                method: 'POST',
                url: '/application/reauthenticate/',
                data: {
                    email: email,
                    password: password
                }
            });
            const credentials = await secure.get('credentials');
            if (email !== credentials.email && password !== credentials.password) {
                setRequestError('Invalid email or password');
            } else {
                setEmail('');
                setPassword('');
                forgotPin();
                setShowModal(false);
            }

        } catch (error) {
            if (error.response && error.response.status === 401) {
                setRequestError('Invalid email or password.');
            } else {
                setRequestError('An unexpected error ocurred. Please try again.');
                console.log(error);
            }
        }

    }

    return (
        <View style={styles.container}>
            {loading ? (
                <Text style={styles.messageText}>Loading</Text>
            ) : (
                <View style={styles.message}>
                    <Text style={styles.messageText}>
                        {message}
                    </Text>
                </View>
            )}

            <View style={styles.onRow}>
                {[...Array(pinLength).keys()].map(i => {
                    const isSelected = i < pin.length;
                    return <MotiView
                        key={`pin-${i}`}
                        style={styles.pinArray}
                        transition={{ type: 'timing', duration: 100 }}
                        animate={{ height: isSelected ? pinTextSize : 2, marginBottom: isSelected ? pinTextSize : 0, translateX: matchedPins ? [0, -20, 20, -20, 20, 0] : 0 }} />;
                })}
            </View>
            <PinPad onPress={handlePress} />
            <View style={styles.onRowForgotPin}>
                <Text>Forgot your Pin?</Text>
                <TouchableOpacity>
                    <Text style={styles.forgotPin} onPress={showModalOnScreen}> Reset</Text>
                </TouchableOpacity>
            </View>

            <Modal animationType="slide" transparent={true} visible={showModal} onRequestClose={() => setShowModal(false)}>
                <BlurView
                    style={StyleSheet.absoluteFill}
                    blurType="light"
                    blurAmount={10}
                    reducedTransparencyFallbackColor="white" />
                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                    <Input
                        title="Email"
                        value={email}
                        error={emailError}
                        setValue={setEmail}
                        setError={setEmailError}
                        clear={setRequestError} />
                    <Input
                        title="Password"
                        value={password}
                        error={passwordError}
                        setValue={setPassword}
                        setError={setPasswordError}
                        clear={setRequestError} />
                    {requestError ? (
                        <Text style={styles.errorText}>{requestError}</Text>
                    ) : null}
                    <CustomButton title="Reset PIN" onPress={resetPIN} />
                    <CustomButton title="Close" onPress={closeModal} />
                </View>
            </Modal>

        </View>
    );
}

// Styles for Pin component
const styles = StyleSheet.create({
    // Main container
    container: {
        flex: 1, // Take up all of the space
        justifyContent: 'center',
        alignItems: 'center',
    },
    // Styles for digit circles
    circles: {
        width: circleSize,
        height: circleSize,
        borderRadius: circleSize,
        borderColor: 'black',
        alignItems: 'center',
        justifyContent: 'center',
    },
    // Styles for text in digit circles
    circleText: {
        fontSize: circleTextSize,
        color: '#000000'
    },
    // Styles for pin array
    pinArray: {
        width: pinTextSize,
        borderRadius: pinTextSize,
        backgroundColor: 'red',
    },
    // Styles for pin entry row
    onRow: {
        flexDirection: 'row',
        gap: 15,
        marginBottom: 40,
        height: pinTextSize * 2,
        alignItems: 'flex-end'
    },
    // Styles for pin setup message
    message: {
        marginBottom: 20,
    },
    // Styles for pin setup message text
    messageText: {
        fontSize: 20,
        color: '#000000'
    },
    // Styles for forgot pin text
    forgotPin: {
        color: '#00008b',
    },
    // Styles for row containing forgot pin text
    onRowForgotPin: {
        flexDirection: 'row',
        marginTop: 20,
    },
    // Styles for error text
    errorText: {
        color: '#ff0000',
        fontSize: 14,
        textAlign: 'center',
        width: '100%',
        marginTop: '3%',
    },
});
