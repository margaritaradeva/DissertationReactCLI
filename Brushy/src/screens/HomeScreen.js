import { createMaterialBottomTabNavigator } from '@react-navigation/material-bottom-tabs'; // Bottom tabs navigator from Material design
import { Provider } from 'react-native-paper'; // Provide theme context to the react-native -paper components
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'; // Icons
import Home from './Home';
import RewardsScreen from './RewardsScreen';
import ParentScreen from './ParentScreen';
// Create the bottom tab navigator (menu) 
// Docs available at https://reactnavigation.org/docs/material-bottom-tab-navigator
const Tab = createMaterialBottomTabNavigator();

export default function HomeScreen() {
  return (
   
    // The provider component will enable theme usage accross all of the screens (components)
    <Provider>
     
        {/* Tab.Navigator - tell the app what the navigation structure of the bottom tabs is */}
        <Tab.Navigator
          screenOptions={{
            tabBarActiveTintColor: 'pink',
          }}>
          {/* Tab.Screen - defines each individual tab item which directs the user to a screen */}
          <Tab.Screen
            name="Homee"
            component={Home}
            options={{
              tabBarIcon: ({ color }) => (
                <MaterialCommunityIcons name="toothbrush" color={color} size={26} />
              ),
            }}
          />
          <Tab.Screen
            name="Rewards"
            component={RewardsScreen}
            options={{
              tabBarIcon: ({ color }) => (
                <MaterialCommunityIcons name="star" color={color} size={26} />
              ),
            }}
          />
          <Tab.Screen
            name="Parents"
            component={ParentScreen}
            options={{
              tabBarIcon: ({ color }) => (
                <MaterialCommunityIcons name="account-switch-outline" color={color} size={26} />
              ),
            }}
          />
        </Tab.Navigator>
    </Provider>
  );}