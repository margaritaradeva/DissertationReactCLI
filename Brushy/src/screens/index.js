// src/screens/index.js

import HomeScreen from './HomeScreen';
import RewardsScreen from './RewardsScreen';
import ParentScreen from './ParentScreen'
import SignIn from './SignIn';
import SignUp from './SignUp';
import Message from './Message';
import Search from './Search';
import LoadingScreen from './LoadingScreen';

export {
  HomeScreen,
  RewardsScreen,
  ParentScreen,
  LoadingScreen,
  SignIn,
  SignUp,
  Message,
  Search
};
