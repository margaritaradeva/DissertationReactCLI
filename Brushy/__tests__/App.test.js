import React from 'react';
import renderer from 'react-test-renderer';
import { HomeScreen, ParentScreen, RewardsScreen } from '../src/screens';

// Make sure the screens render correctly
test('renders HomeScreen correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders ParentScreen correctly', () => {
  const tree = renderer.create(<ParentScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders RewardsScreen correctly', () => {
  const tree = renderer.create(<RewardsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
