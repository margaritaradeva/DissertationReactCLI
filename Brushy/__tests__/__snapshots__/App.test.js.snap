// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomeScreen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Home!
  </Text>
</View>
`;

exports[`renders ParentScreen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Parents!
  </Text>
</View>
`;

exports[`renders RewardsScreen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Home!
  </Text>
</View>
`;
